src/ucp/api/ucpx.h

```diff
@@ -19,6 +19,139 @@
 
 BEGIN_C_DECLS
 
+#define UCP_FEATURE_AM UCP_FEATURE_X
+
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Callback to process incoming active message
+ *
+ * When the callback is called, @a flags indicates how @a data should be handled.
+ *  
+ * @param [in]  arg      User-defined argument.
+ * @param [in]  data     Points to the received data. This data may
+ *                       persist after the callback returns and need
+ *                       to be freed with @ref ucp_am_data_release
+ * @param [in]  length   Length of data.
+ * @param [in]  reply_ep If the active message is sent with the 
+ *                       UCP_AM_SEND_REPLY flag, the sending ep
+ *                       will be passed in. If not, NULL will be passed
+ * @param [in]  flags    If this flag is set to UCP_CB_PARAM_FLAG_DATA,
+ *                       the callback can return UCS_INPROGRESS and
+ *                       data will persist after the callback returns
+ *
+ * @return UCS_OK        @a data will not persist after the callback returns
+ *                      
+ * @return UCS_INPROGRESS Can only be returned if flags is set to
+ *                        UCP_CB_PARAM_FLAG_DATA. If UCP_INPROGRESS
+ *                        is returned, data will persist after the
+ *                        callback has returned. To free the memory,
+ *                        a pointer to the data must be passed into
+ *                        @ref ucp_am_data_release
+ *
+ * @note This callback could be set and released
+ *       by @ref ucp_worker_set_am_handler function.
+ *
+ */
+typedef ucs_status_t (*ucp_am_callback_t)(void *arg, void *data, size_t length,
+                                          ucp_ep_h reply_ep, unsigned flags);
```

Comment:
BTW, how user should handle `reply_ep`? There are several cases:
1) User had created the EP before and UCP passes the same handle to this callback, right?
2) User hadn't created the EP and UCP should create `reply_ep` internally. Should user close this EP? What if user will create another EP connected to the originator after this callback, should that EP be `reply_ep`?
3) Is it allowed to use `reply_ep` outside of the callback?